core/src/main/java/hudson/model/AbstractProject.java

```diff
@@ -1207,7 +1207,14 @@ public boolean checkout(AbstractBuild build, Launcher launcher, BuildListener li
             return true;    // no SCM
 
         FilePath workspace = build.getWorkspace();
-        workspace.mkdirs();
+        if(workspace!=null){
+            workspace.mkdirs();
+        }
+        else{
+            LOGGER.log(Level.SEVERE,"Workspace does not exist!!");
```

Comment:
This log message is not particularly useful.

* Never use multiple exclamation points, or I shall be forced to quote Terry Pratchett 😉 In fact, it's fine to use none at all.
* Which agent? (`build.getBuiltOn()` perhaps?)
  * Perhaps it would be useful to include the status of the agent, if safely feasible?
* Which project?
* In fact, it's even a misleading statement: The workspace is perhaps _undefined_, as Jenkins _creates_ it if defined a few lines up. So the problem isn't existence (on the file system).
* It's not a severe error of the system. `WARNING` should be fine.

With the current lack of information, the log message is pretty much useless.

Reply:
The log message should also go to `BuildListener listener` IMHO. In such case there will be no need to specify the build.

Reply:
No logging is needed at all. Just throw an `AbortException`.